lib/setupmanagers/hckclient.rb

```diff
@@ -18,23 +18,26 @@ def initialize(setup_manager, scope, studio, name, run_opts)
       @studio = studio
       @name = name
       @kit = setup_manager.kit
-      @pool = 'Default Pool'
       @logger.info("Starting client #{name}")
       @runner = setup_manager.run_client(scope, @name, run_opts)
       scope << self
     end
 
+    def pool
+      machine_in_working_pool? ? @project.engine_tag : 'Default Pool'
```

Comment:
You can instead use `@studio.list_pools` and return the pool containing this machine.